test/tests/pbs_admin_suspend.py

```diff
@@ -405,3 +413,416 @@ def test_degraded_resv(self):
+
+        # set foo on vn[1]
+        self.server.manager(
+            MGR_CMD_SET, NODE, {'resources_available.foo': 5}, id='vn[1]',
+            runas=ROOT_USER)
+
+        # set foo in sched_config
+        self.scheduler.add_resource('foo')
+
+        # submit a few jobs
+        j = Job(TEST_USER)
+        j.set_attributes({'Resource_List.select': 'vnode=vn[1]'})
+        jid1 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # admin suspend the job to put the node to maintenance
+        self.server.sigjob(jid1, 'admin-suspend', runas=ROOT_USER)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[1]')
+
+        # submit other jobs asking for specific resources on vn[1]
+        j = Job(TEST_USER)
+        j.set_attributes({'foo': '2'})
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        # submit more jobs. They should be running
+        j = Job(TEST_USER)
+        jid3 = self.server.submit(j)
+        jid4 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid4)
+
+        # verify that vn[1] is still in maintenance and
+        # job3 and job4 not running on vn[1]
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[1]')
+        try:
+            self.server.expect(
+                JOB, {'exec_vnode': 'vn[1]'}, id=jid3, max_attempts=20)
+            self.server.expect(
+                JOB, {'exec_vnode': 'vn[1]'}, id=jid4, max_attempts=20)
+        except Exception, e:
+            self.assertFalse(e.rv)
+            print ("jid3 and jid4 not running on vn[1] as expected")
+
+    def test_list_jobs_1(self):
+        """
+        Test to list and set maintenance_jobs as various users
+        """
+
+        # submit a few jobs
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        jid2 = self.server.submit(j)
+        jid3 = self.server.submit(j)
+
+        # verify that all are running
+        self.server.expect(JOB, {'job_state=R': 3, 'substate=42': 3})
+
+        # admin-suspend 2 of them
+        self.server.sigjob(jid2, 'admin-suspend', runas=ROOT_USER)
+        self.server.sigjob(jid3, 'admin-suspend', runas=ROOT_USER)
+
+        # node state is in maintenance
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[0]')
+
+        # list maintenance_jobs as root
+        self.server.expect(NODE, {'maintenance_jobs': jid2 + "," + jid3},
+                           runas=ROOT_USER)
+
+        # list maintenance jobs as user
+        self.server.expect(NODE, {'maintenance_jobs': jid2 + "," + jid3},
+                           runas=TEST_USER)
+
+        # set an operator
+        self.server.manager(MGR_CMD_SET, SERVER, {'operators': 'pbsoper@*'})
```

Comment:
OPER_USER was identified as pbsoper but it was not set automatically in server so I am getting different error message. I would leave it as it is for this test.